anomalydetection/hampel.py

```diff
@@ -58,3 +59,39 @@ def validate_arguments(window_size, threshold):
     else:
         if threshold < 0:
             raise ValueError("Threshold must be positive.")
+
+
+@jit(nopython=True)
+def detect_using_numba(time_series, window_size, threshold=3, k=GAUSSIAN_SCALE_FACTOR):
+    """
+    Hampel filter implementation that works on numpy arrays, implemented with numba. Snatched from this implementation:
+    https://github.com/erykml/medium_articles/blob/master/Machine%20Learning/outlier_detection_hampel_filter.ipynb
```

Comment:
There is no license on this repo, I would be wary of using anything from it in our code, see https://opensource.stackexchange.com/questions/1720/what-can-i-assume-if-a-publicly-published-project-has-no-license

Reply:
I have found the implementation many places and created my own version with a lot of variables changed. There is also very limited ways of writing these 5 lines efficiently in numpy, so I think we are ok. I removed the link if it creates license problems and does not help the reader as intended.